content/rust/github/publish-github-release-multi-arch.md

```diff
@@ -0,0 +1,419 @@
+{{< / alert >}}
+
+# Cross Compiling the Rust Project
+
+The goal is to compile the Rust project for three different platforms: `amd64`, `aarch64` and `armv7`.
+The Cross-Compilation Docker image from the previous tutorial supports only the current platform (`amd64` in this case) and `aarch64`.
+To add support for `armv7` you need to install a compatible linker and the Rust Standard Library.
+Add the highlighted lines to the `dockerfile` and build it.
+
+{{< highlight text "linenos=table,hl_lines=5-6 8" >}}
+FROM rust:latest
+RUN apt-get update \
+ && apt-get install -y --no-install-recommends g++-aarch64-linux-gnu libc6-dev-arm64-cross \
+ && rustup target add aarch64-unknown-linux-gnu \
+ && apt-get install -y --no-install-recommends g++-arm-linux-gnueabihf libc6-dev-armhf-cross \
+ && rustup target add armv7-unknown-linux-gnueabihf
+ENV CARGO_TARGET_AARCH64_UNKNOWN_LINUX_GNU_LINKER=aarch64-linux-gnu-gcc
+ENV CARGO_TARGET_ARMV7_UNKNOWN_LINUX_GNUEABIHF_LINKER=arm-linux-gnueabihf-gcc
+{{< / highlight >}}
+
+```console
+$ docker build -t rust-multiarch .
+```
+
+The usage is identical but with support for `armv7` now:
+
+```console
+# Build amd64 (default)
+$ docker run --rm -v <path to your project>:/rust-project -w /rust-project rust-multiarch cargo build
+# Build aarch64
+$ docker run --rm -v <path to your project>:/rust-project -w /rust-project rust-multiarch cargo build --target=aarch64-unknown-linux-gnu
+# Build armv7
+$ docker run --rm -v <path to your project>:/rust-project -w /rust-project rust-multiarch cargo build --target=armv7-unknown-linux-gnueabihf
+```
+
+This creates three compiled binaries:
+
+```console
+target/debug/hello-world
+target/aarch64-unknown-linux-gnu/debug/hello-world
+target/armv7-unknown-linux-gnueabihf/debug/hello-world
+```
+
+To organize these assets create a new directory `assets` and copy (and rename) the binaries:
+
+```console
+$ mkdir assets
+$ cp target/debug/hello-world assets/hello-world-amd64
+$ cp target/aarch64-unknown-linux-gnu/debug/hello-world assets/hello-world-aarch64
+$ cp target/armv7-unknown-linux-gnueabihf/debug/hello-world assets/hello-world-armv7
+```
+
+# Creating a GitHub Release
+
+To create a GitHub release, you first need to [commit our project to GitHub](https://help.github.com/en/articles/creating-a-repository-on-github) and create a git tag for the current version:
+
+```console
+$ git tag v1.0.0
+$ git push origin v1.0.0
+```
+
+To create a release from this tag you could use the GitHub user interface or the GitHub API.
+This tutorial shows the usage of the API with `curl`.
+
+{{< alert info >}}
+To use the GitHub API you need [a token for authorization](https://github.com/settings/tokens).
+If you don't have one, create it now.
+{{< / alert >}}
+
+First you need to [create a release](https://developer.github.com/v3/repos/releases/#create-a-release) for the tag:
```

Comment:
Does it really make sense to describe the manual path of creating a release? I mean we are writing guides for Drone, so why the manual approach at all?

Reply:
The manual path was a requirement.

Reply:
I think this is a good example of manually creating the release with the user interface, which is going to be a more common approach than API requests
https://tutorials.drone.io/create-multi-arch-github-release-go-binary/#create-a-github-release

Reply:
Should I change it to the gui approach too? I used the api so that it isn't just the same as in the other tutorial.